tests/conftest.py

```diff
@@ -145,9 +145,13 @@ def install_signal_handlers(self) -> None:
             loop.create_task(super().serve(sockets=sockets)),
             loop.create_task(self.watch_restarts()),
         }
-
         await asyncio.wait(tasks)
 
+    @property
+    def url(self) -> URL:
+        protocol = "https" if self.config.is_ssl else "http"
+        return URL(f"{protocol}://{self.config.host}:{self.config.port}/")
```

Comment:
Just a nit: could you consider placing this property at the top of the class? The `serve`,  `restart` and `watch_restarts()` methods are all kind of related in a non-immediately obvious way, so I think it's best to keep them together.

Reply:
Done